File: src/main/java/com/bajdcc/LALR1/grammar/runtime/service/RuntimeService.java
package com.bajdcc.LALR1.grammar.runtime.service;

import com.bajdcc.LALR1.grammar.runtime.RuntimeProcess;

/**
 * 【运行时】运行时服务
 *
 * @author bajdcc
 */
public class RuntimeService implements IRuntimeService {

	private RuntimeProcess process;
	private RuntimePipeService pipe;
	private RuntimeShareService share;
	private RuntimeFileService file;
	private RuntimeDialogService dialog;

	public RuntimeService(RuntimeProcess process) {
		this.process = process;
		this.pipe = new RuntimePipeService(this);
		this.share = new RuntimeShareService(this);
		this.file = new RuntimeFileService(this);
		this.dialog = new RuntimeDialogService(this);
	}

	@Override
	public IRuntimePipeService getPipeService() {
		return pipe;
	}

	@Override
	public IRuntimeShareService getShareService() {
		return share;
	}

	@Override
	public IRuntimeProcessService getProcessService() {
		return process;
	}

	@Override
	public IRuntimeFileService getFileService() {
		return file;
	}

	@Override
	public IRuntimeDialogService getDialogService() {
		return dialog;
	}
}

File: src/main/java/com/bajdcc/LALR1/grammar/runtime/service/RuntimeService.kt
package com.bajdcc.LALR1.grammar.runtime.service

import com.bajdcc.LALR1.grammar.runtime.RuntimeProcess

/**
 * 【运行时】运行时服务
 *
 * @author bajdcc
 */
class RuntimeService(private val process: RuntimeProcess) : IRuntimeService {
    private val pipe: RuntimePipeService = RuntimePipeService(this)
    private val share: RuntimeShareService = RuntimeShareService(this)
    private val file: RuntimeFileService = RuntimeFileService(this)
    private val dialog: RuntimeDialogService = RuntimeDialogService(this)
    private val user: RuntimeUserService = RuntimeUserService(this)

    override val pipeService: IRuntimePipeService
        get() = pipe

    override val shareService: IRuntimeShareService
        get() = share

    override val processService: IRuntimeProcessService
        get() = process

    override val fileService: IRuntimeFileService
        get() = file

    override val dialogService: IRuntimeDialogService
        get() = dialog

    override val userService: IRuntimeUserService
        get() = user
}
